src/lib/Libifl/pbsD_connect.c

```diff
@@ -357,63 +248,46 @@ __pbs_connect_extend(char *server, char *extend_data)
 			setenv("SystemRoot", "C:\\WINDOWS", 1);
 		}
 #endif
-		sock = socket(AF_INET, SOCK_STREAM, 0);
+	sd = socket(AF_INET, SOCK_STREAM, 0);
+	if (sd == -1) {
+		pbs_errno = PBSE_SYSTEM;
+		return -1;
+	}
 
+	pbs_strncpy(pbs_server, server, sizeof(pbs_server)-1); /* set for error messages from commands */
+	pbs_server[sizeof(pbs_server) - 1] = '\0';
 		/* and connect... */
 
-		if (have_alt) {
-			server = altservers[i];
-		}
-		strcpy(pbs_server, server); /* set for error messages from commands */
-
-		/* If a specific host name is defined which the client should use */
-
-		if (pbs_conf.pbs_public_host_name) {
-			/* my address will be in my_sockaddr,  bind the socket to it */
-			my_sockaddr.sin_port = 0;
-			if (bind(sock, (struct sockaddr *)&my_sockaddr, sizeof(my_sockaddr)) != 0) {
-				return -1;
-			}
-		}
-
-		if (get_hostsockaddr(server, &server_addr) != 0)
+	/* If a specific host name is defined which the client should use */
+	if (pbs_conf.pbs_public_host_name) {
+		if (get_hostsockaddr(pbs_conf.pbs_public_host_name, &my_sockaddr) != 0)
```

Comment:
I feel we can keep tcp_connect() devoid of any pollution from pbs_conf. The goal for tcp_connect() can be very simple, given the hostname and port, connect to it, do not try to interpret it any further than what is provided. The callers of this function should decide what actual hostname should be passed to it (and how pbs_conf.pbs_public_host_name) affects what is passed to this routine.